tests/update_assertions.py

```diff
@@ -0,0 +1,282 @@
+    if match:
+        return match.group(1)
+    return None
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Update rule assertions in YAML files (interactive or via flags)."
+    )
+    parser.add_argument("--rule",
+                        help="Exact rule name to search for (underscores will be turned into dashes).")
+    parser.add_argument("--bulk", action="store_true",
+                        help="If specified, apply updates in bulk to all matching assertions.")
+    parser.add_argument("--default-result",
+                        help="Set this as the new default_result for matching rules.")
+    parser.add_argument("--result-after-remediation",
+                        help="Set this as the new result_after_remediation for matching rules.")
+    parser.add_argument("--exclude-version", action="append", default=[],
+                        help="Exclude certain OCP versions from changes. Can be used multiple times.")
+    args = parser.parse_args()
+
+    # 1) Get the rule name (either from CLI or prompt)
+    if args.rule:
+        rule_name = args.rule.strip()
+    else:
+        rule_name = input("Enter the rule name to search for: ").strip()
+
+    if not rule_name:
+        print("No rule name provided. Exiting.")
+        sys.exit(0)
+
+    # Replace underscores with dashes in the user rule.
+    # We only match if the normalized key ends with this string.
+    rule_name = rule_name.replace("_", "-")
+
+    # Path to your assertion files
+    assertions_dir = "assertions/ocp4"
+
+    # Collect excluded versions
+    exclude_versions = set(args.exclude_version) if args.exclude_version else set()
+    if exclude_versions:
+        print(f"\nExcluding OCP versions: {', '.join(sorted(exclude_versions))}")
+
+    # Find all YAML files
+    yaml_files = glob.glob(os.path.join(assertions_dir, "**/*.yml"), recursive=True)
+
+    # Collect all matches: (yaml_file, key_in_yaml, default_res, remediation_res, version)
+    matches = []
+    for yf in yaml_files:
+        # Extract the OCP version from the filename
+        file_version = extract_ocp_version(yf)
+        # If version is in the excluded set, skip entirely
+        if file_version and file_version in exclude_versions:
+            continue
+
+        data = load_yaml(yf)
+        rule_results = data.get("rule_results", {})
+        if not isinstance(rule_results, dict):
+            continue
+
+        # For each key, check if it ends with rule_name after normalizing
+        for key, val in rule_results.items():
+            normalized_key = key.replace("_", "-")
+            if normalized_key.endswith(rule_name):
+                dres = val.get("default_result")
+                rres = val.get("result_after_remediation")
+                matches.append((yf, key, dres, rres, file_version))
+
+    if not matches:
+        print(f"\nNo matching rule_results found that end with '{rule_name}' "
```

Comment:
Something that might be nice here would be to give users the option to add new assertions.